make skaffold is broken

The `README.md` we provide mentions at least three `skaffold` related calls, `make skaffold`, `make skaffold MODE=run` and `make e2e`. The first two are broken, at least on MacOS:

```
$ make skaffold
make: Circular skaffold <- skaffold dependency dropped.
make: Nothing to be done for `skaffold'.
```

```
$ make skaffold MODE=run
skaffold run \
		-f /Users/pires/work/virtual-kubelet/hack/skaffold/virtual-kubelet/skaffold.yml \
		-p local
Generating tags...
 - virtual-kubelet -> virtual-kubelet:v0.11.1-6-gcd0abc4d-dirty
Tags generated in 22.50431ms
Starting build...
Found [minikube] context, using local docker daemon.
Building [virtual-kubelet]...
Sending build context to Docker daemon

FATA[0000] failed to build: build failed: building [virtual-kubelet]: build artifact: docker build: error during connect: Post https://192.168.64.3:2376/v1.38/build?buildargs=null&cachefrom=null&cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=hack%2Fskaffold%2Fvirtual-kubelet%2FDockerfile&forcerm=1&labels=null&memory=0&memswap=0&networkmode=&rm=0&shmsize=0&t=virtual-kubelet%3Av0.11.1-6-gcd0abc4d-dirty&target=&ulimits=null: creating docker context: getting relative tar paths: file pattern [bin/e2e/virtual-kubelet] must match at least one file
make: *** [skaffold/run] Error 1
```

@cpuguy83 this is the result of the changes in #564. Can you, please, fix, so that the commands above work again? Thank you!

--- a/Makefile.e2e
+++ b/Makefile.e2e
@@ -11,7 +11,8 @@ skaffold.validate:
 # MODE must be set to one of "dev" (default), "delete" or "run", and is used as the skaffold command to be run.
 .PHONY: skaffold
 skaffold: MODE ?= dev
-skaffold: skaffold/$(MODE)
+.SECONDEXPANSION:
+skaffold: skaffold/$$(MODE)
 
 .PHONY: skaffold/%
 skaffold/%: PROFILE := local
@@ -19,6 +20,8 @@ skaffold/%: skaffold.validate
 	skaffold $(*) \
 		-f $(PWD)/hack/skaffold/virtual-kubelet/skaffold.yml \
 		-p $(PROFILE)
+
+skaffold/run skaffold/dev: bin/e2e/virtual-kubelet
 
 bin/e2e:
 	@mkdir -p bin/e2e

